types/aws-greengrass-core-sdk/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for aws-greengrass-core-sdk 1.6
+// Project: https://github.com/aws/aws-greengrass-core-sdk-js#readme
+// Definitions by: Alister Hatt <https://github.com/ozxcorp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+export interface PublishParams {
+    topic: string;
+    payload: Buffer | Blob | string | ArrayLike<any>;
+    queueFullPolicy?: 'AllOrError' | 'BestEffort';
+}
+
+export interface IotCallbackData {
+    payload: Buffer | Blob | string | ArrayLike<any>;
+}
+
+export type IotCallback = (error: Error | null, data: IotCallbackData | null) => void;
+
+export interface IotDataClient {
```

Comment:
I'm not sure I understand what this adds.  Couldn't it be replaced by `IotData` at any use site?

Reply:
You're correct. I had defined that interface in my use project for unit/integration testing, but it doesn't really belong in the definition library as clients could easily implement their own interface to the concrete class as required. 

I'll make an update.

Reply:
Done. Not sure why CI is failing though. :(

Reply:
The CI failure is unrelated.  Thanks for the update!